### ISSUE TITLE
shellhub gateway not working on Old CPU

### ISSUE THREAD

**Issue by nejeoui** [shellhub-io/shellhub#737]
After installing shellhub and starting the the containers using docker-compse i got the error message on the console

c./bin/docker-compose up
shellhub_mongo_1 is up-to-date
shellhub_ssh_1 is up-to-date
shellhub_api_1 is up-to-date
shellhub_ui_1 is up-to-date
Starting shellhub_gateway_1 ... done
Attaching to shellhub_mongo_1, shellhub_ssh_1, shellhub_api_1, shellhub_ui_1, shellhub_gateway_1
api_1      |
api_1      |    ____    __
api_1      |   / __/___/ /  ___
api_1      |  / _// __/ _ \/ _ \
api_1      | /___/\__/_//_/\___/ v3.3.10-dev
api_1      | High performance, minimalist Go web framework
api_1      | https://echo.labstack.com
api_1      | ____________________________________O/_______
api_1      |                                     O\
api_1      | ⇨ http server started on [::]:8080
mongo_1    | 2021-02-24T14:48:50.370+0000 I  COMMAND  [conn3] CMD: dropIndexes main.users: "tenant_id"
mongo_1    | 2021-02-24T14:48:50.403+0000 I  COMMAND  [conn3] CMD: dropIndexes main.users: "session_record"
mongo_1    | 2021-02-24T14:53:32.846+0000 I  SHARDING [LogicalSessionCacheReap] Marking collection config.transactions as collection version: <unsharded>
shellhub_gateway_1 exited with code 132
shellhub_gateway_1 exited with code 132
shellhub_gateway_1 exited with code 132

After installing shellhub and starting the the containers using docker-compse i got the error message on the console
```
c./bin/docker-compose up
shellhub_mongo_1 is up-to-date
shellhub_ssh_1 is up-to-date
shellhub_api_1 is up-to-date
shellhub_ui_1 is up-to-date
Starting shellhub_gateway_1 ... done
Attaching to shellhub_mongo_1, shellhub_ssh_1, shellhub_api_1, shellhub_ui_1, shellhub_gateway_1
api_1      |
api_1      |    ____    __
api_1      |   / __/___/ /  ___
api_1      |  / _// __/ _ \/ _ \
api_1      | /___/\__/_//_/\___/ v3.3.10-dev
api_1      | High performance, minimalist Go web framework
api_1      | https://echo.labstack.com
api_1      | ____________________________________O/_______
api_1      |                                     O\
api_1      | ⇨ http server started on [::]:8080
mongo_1    | 2021-02-24T14:48:50.370+0000 I  COMMAND  [conn3] CMD: dropIndexes main.users: "tenant_id"
mongo_1    | 2021-02-24T14:48:50.403+0000 I  COMMAND  [conn3] CMD: dropIndexes main.users: "session_record"
mongo_1    | 2021-02-24T14:53:32.846+0000 I  SHARDING [LogicalSessionCacheReap] Marking collection config.transactions as collection version: <unsharded>
shellhub_gateway_1 exited with code 132
shellhub_gateway_1 exited with code 132
shellhub_gateway_1 exited with code 132
````
It seems that shellhub_gateway use AVX(Advanced Vector Extensions) which is not supported on my old intel atom CPU.

Any idea how to get shellhub work on old CPUs ?
[stack overflow link](https://stackoverflow.com/questions/66353779/how-to-get-shellhub-io-working-on-old-cpu)

**Comment by nejeoui** [comment]
CPU Detail : Model name:                      Intel(R) Atom(TM) CPU N2800   @ 1.86GHz

**Comment by gustavosbarreto** [comment]
In the gateway service we are just using the upstream openresty image. So we support whatever they do, which apparently no longer includes support for 10+ year old hardware (which does not have SSE 4.2).

**Comment by gustavosbarreto** [comment]
@nejeoui Can you please follow these steps to see if these work, if not, please let me know:

1) Replace `FROM` instruction value of the `gateway/Dockerfile` with `openresty/openresty:1.15.8.2-6-alpine-nosse42`
2) Manually build the image: `docker build -t nejeoui/gateway:1 -f gateway/Dockerfile .`
3) Replace `image` value from `gateway` service in docker-compose with: `nejeoui/gateway:1`

**Comment by nejeoui** [comment]
@gustavosbarreto thank you for your quick response, i really appreciate your reactivity.

1. I have build manually the local image as advised  
2. and replaced the image in docker-compose 
```
root@ns3002211:~# docker ps
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                    NAMES
0ad2f9befe69        nejeoui/gateway:1       "/entrypoint.sh /usr…"   6 minutes ago       Up 4 minutes                                 shellhub_gateway_1
2f5b7f3871c1        shellhubio/ui:v0.5.2    "/scripts/entrypoint…"   7 minutes ago       Up 4 minutes        80/tcp                   shellhub_ui_1
0280c1219547        shellhubio/api:v0.5.2   "/bin/sh -c /api"        7 minutes ago       Up 4 minutes                                 shellhub_api_1
8859d267c3c8        mongo:4.2.12            "docker-entrypoint.s…"   7 minutes ago       Up 4 minutes        27017/tcp                shellhub_mongo_1
b18b257fb092        shellhubio/ssh:v0.5.2   "/bin/sh -c /ssh"        7 minutes ago       Up 4 minutes        0.0.0.0:2222->2222/tcp   shellhub_ssh_1
```

as you can see the port 8090 is not mapped (i have configured the port 8090 for HTTP)

**Comment by gustavosbarreto** [comment]
> @gustavosbarreto thank you for your quick response, i really appreciate your reactivity.
> 
> 1. I have built manually the local image as advised
> 2. and replaced the image in docker-compose
> 
> ```
> root@ns3002211:~# docker ps
> CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                    NAMES
> 0ad2f9befe69        nejeoui/gateway:1       "/entrypoint.sh /usr…"   6 minutes ago       Up 4 minutes                                 shellhub_gateway_1
> 2f5b7f3871c1        shellhubio/ui:v0.5.2    "/scripts/entrypoint…"   7 minutes ago       Up 4 minutes        80/tcp                   shellhub_ui_1
> 0280c1219547        shellhubio/api:v0.5.2   "/bin/sh -c /api"        7 minutes ago       Up 4 minutes                                 shellhub_api_1
> 8859d267c3c8        mongo:4.2.12            "docker-entrypoint.s…"   7 minutes ago       Up 4 minutes        27017/tcp                shellhub_mongo_1
> b18b257fb092        shellhubio/ssh:v0.5.2   "/bin/sh -c /ssh"        7 minutes ago       Up 4 minutes        0.0.0.0:2222->2222/tcp   shellhub_ssh_1
> ```
> 
> as you can see the port 8090 is not mapped (i have configured the port 8090 for HTTP)

It seems that gateway service is up and running but no ports are mapped.

**Comment by nejeoui** [comment]
Sure
here it is 
```
root@ns3002211:~/shellhub# ./bin/docker-compose logs gateway
Attaching to shellhub_gateway_1
root@ns3002211:~/shellhub#
```

**Comment by gustavosbarreto** [comment]
Looks good, no errors.

It seems it's fixed in 8a147071e21cbd47d6bcf86827a1f9286d184f0d.

You can create a `docker-compose.override.yml` with following config to workaround the issue until we release next version with a fix:

```
version: '3.7'

services:
  gateway:
    ports:
      - "${SHELLHUB_HTTP_PORT}:80"
```

**Comment by nejeoui** [comment]
Thanks that solve my problem
